### DISCUSSION TITLE
TextColorToGenerator - Make access modifiers for generated classes for Maui Controls adjustable

### DISCUSSION THREAD

**Discussion by loop8ack** [CommunityToolkit/Maui#769] [New Feature Discussions]
This discussion is based on my question: #756
I have a solution for my project, but the real problem is still there.

I had the problem that the TextColorToGenerator always generates the code for Maui controls `public`, which causes conflicts with multiple projects.
Since I always set TreatWarningsAsErrors to true, I clearly noticed this (many CS0436 erros), others will see this as warnings.

```xml
<PropertyGroup>
    <CommunityToolkit_Maui_GeneratedCodeForMauiControlsPublic>true</CommunityToolkit_Maui_GeneratedCodeForMauiControlsPublic>
</PropertyGroup>
```

The purpose of this property is to be able to provide a simple solution in case of a problem with this change (`public` -> `internal`).
Accepted values are `true` (`public`) or `false` (`internal`), the default value would be empty (`internal`).
Or we change this setting so that the default value is `public` and can be changed to `internal`.

It would also be enough to always leave the access modifier `internal` (without setting), but that would be a breaking change, with no way to do anything about it.

What do you think?

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
